distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso

```diff
@@ -209,116 +210,77 @@ Text.characters self =
-
-## ALIAS find_all
+         example_find =
+             ## This matches `abc` @ character 11
+             "aabbbbccccaabcaaaa".find "a[ab]c"
+         example_find_insensitive =
+             ## This matches `aBc` @ character 11
+             "aabbbbccccaaBcaaaa".find "a[ab]c" Case_Sensitivity.Insensitive
+Text.find : Text -> Case_Sensitivity -> Match | Nothing ! Compile_Error
+Text.find self pattern=".*" case_sensitivity=Case_Sensitivity.Sensitive =
+    case_insensitive = case_sensitivity.is_case_insensitive
+    Regex.compile pattern case_insensitive=case_insensitive . match self Matching_Mode.First
 
-   Matches all occurrences text in `self` against the provided `term`, returning
-   a vector of matches.
+## Finds all the matches of the regular expression `pattern` in `self`,
+   returning a Vector. If not found, will be an empty Vector.
 
    Arguments:
-   - term: The pattern to match `self` against. We recommend using _raw text_
-     to write your patterns.
-   - matcher: If a `Text_Matcher`, the text is compared using case-sensitivity
-     rules specified in the matcher. If a `Regex_Matcher`, the term is used as a
-     regular expression and matched using the associated options.
+   - pattern: The pattern to match `self` against.
+   - case_sensitivity: Specifies if the text values should be compared case
+     sensitively.
 
    > Example
-     Find all substrings matching the regex.
-
-         example_match =
-             regex = "a[ab]c"
-             "aabcbbccaacaa".match regex == ["abc", "aac"]
-Text.match_all : Text -> (Text_Matcher | Regex_Matcher) -> Vector Text
-Text.match_all self term=".*" matcher=Regex_Matcher.Value = case matcher of
-    _ : Text_Matcher ->
-        case_sensitivity = case matcher of
-            Text_Matcher.Case_Sensitive -> Case_Sensitivity.Sensitive
-            Text_Matcher.Case_Insensitive _ -> Case_Sensitivity.Insensitive
-        self.locate_all term case_sensitivity . map .text
-    _ : Regex_Matcher ->
-        case matcher.compile term . match self Regex_Mode.All of
-            Nothing -> []
-            matches -> matches.map m-> m.span 0 . to_grapheme_span . text
+     Find the substring matching the regex.
+
+         example_find_all =
+              ## This matches `aabbbbc` @ character 0 and `abc` @ character 11
+             "aabbbbccccaabcaaaa".find_all "a[ab]+c"
+         example_find_all_insensitive =
+              ## This matches `aABbbbc` @ character 0 and `aBC` @ character 11
+             "aABbbbccccaaBCaaaa".find_all "a[ab]+c" Case_Sensitivity.Insensitive
+Text.find_all : Text -> Case_Sensitivity -> Vector Match ! Compile_Error
+Text.find_all self pattern=".*" case_sensitivity=Case_Sensitivity.Sensitive =
+    case_insensitive = case_sensitivity.is_case_insensitive
+    case Regex.compile pattern case_insensitive=case_insensitive . match self Regex_Mode.All of
+        Nothing -> []
+        matches -> matches
```

Comment:
If we do the check this way, we lose the `locale` information from the `Case_Sensitivity` setting. I understand we cannot easily override the locale setting for Regex - OK. We have the same issue in DB - and I think we should do the same check there too - check if the setting had the default `ROOT` locale - if so - we can proceed. If the user has overridden the locale to something else, they may count on some custom behaviour that we cannot provide here - so I think in that case we should raise an `Illegal_Argument` error saying that custom-locale-backed case insensitivity is not supported here.

Reply:
We are replacing the `Regex.compile` method in the next ticket and it moves to taking a `Case_Sensitive` parameter with exactly that logic.

Reply:
Ah, I did not know this was temporary. Sounds good then!

I think ideally we could add a test case for this too (with this next ticket).